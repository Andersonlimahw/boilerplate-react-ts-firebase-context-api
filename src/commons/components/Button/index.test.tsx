import React from 'react';
import { render, screen } from '@testing-library/react';
import { Button } from './';

test('renders button with correct text', () => {
  render(<Button>Hello</Button>);
  const buttonElement = screen.getByText(/Hello/i);
  expect(buttonElement).toBeDefined();
});

test('applies additional props correctly', () => {
  render(<Button id="myButton" className="custom-class">Click me</Button>);
  const buttonElement = screen.getByText(/Click me/i);

  expect(buttonElement).toMatchSnapshot();
});

// Add more tests as needed
