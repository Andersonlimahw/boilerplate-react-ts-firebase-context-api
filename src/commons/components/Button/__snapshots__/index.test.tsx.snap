// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applies additional props correctly 1`] = `
<button
  class="button"
  id="myButton"
  type="submit"
>
  Click me
</button>
`;
